docs/pages/textarea.js

```diff
@@ -36,219 +53,270 @@ export default function DocsPage({ generatedDocGen }: {| generatedDocGen: DocGen
+        </MainSection.Subsection>
+        <MainSection.Subsection
+          title="With a ref"
+          description={`
+          A \`TextArea\` with an anchor ref to a Popover component
+        `}
+        >
+          <MainSection.Card
+            defaultCode={`
+          function TextAreaPopoverExample() {
+            const [open, setOpen] = React.useState(false);
+            const anchorRef = React.useRef();
+            return (
+              <Box marginBottom={12} width="100%">
+                <TextArea
+                  ref={anchorRef}
+                  label="Focus the TextArea to show the Popover"
+                  id="my-example"
+                  onChange={() => {}}
+                  onBlur={() => setOpen(false)}
+                  onFocus={() => setOpen(true)}
+                />
+                {open && (
+                  <Popover
+                    anchor={anchorRef.current}
+                    idealDirection="down"
+                    onDismiss={() => setOpen(false)}
+                    shouldFocus={false}
+                    size="md"
+                  >
+                    <Box padding={3}>
+                      <Text weight="bold">Example with Popover</Text>
+                    </Box>
+                  </Popover>
+                )}
+              </Box>
+            );
+          }
+          `}
+          />
+        </MainSection.Subsection>
+        <MainSection.Subsection
+          title="With Tags"
+          description={`
     You can include [Tag](/tag) elements in the input using the \`tags\` prop.
 
     Note that the \`TextArea\` component does not internally manage tags. That should be handled in the application state through the component's event callbacks. We recommend creating new tags on enter key presses, and removing them on backspaces when the cursor is in the beginning of the field. We also recommend filtering out empty tags.
 
     This example showcases the recommended behavior.`}
-        defaultCode={`
-function Example(props) {
-  const [value, setValue] = React.useState('');
-  const [tags, setTags] = React.useState(['San Francisco', 'New York']);
-  const ref = React.useRef();
+        >
+          <MainSection.Card
+            defaultCode={`
+          function Example(props) {
+            const [value, setValue] = React.useState('');
+            const [tags, setTags] = React.useState(['San Francisco', 'New York']);
+            const ref = React.useRef();
 
-  const onChangeTagManagement = ({ value }) => {
-    // Create new tags around new lines
-    const tagInput = value.split(/\\n+/);
-    if (tagInput.length > 1) {
-      setTags([
-        ...tags,
-        // Avoid creating a tag on content on the last line, and filter out
-        // empty tags
-        ...tagInput.splice(0, tagInput.length - 1).filter(val => val !== ''),
-      ]);
-    }
-    setValue(tagInput[tagInput.length - 1]);
-  }
+            const onChangeTagManagement = ({ value }) => {
```

Comment:
this looks like a regression:
![Screen Shot 2022-03-31 at 4 29 20 PM](https://user-images.githubusercontent.com/12059539/161165898-1b851c06-239e-4651-adce-968258dae5bc.png)